### PR TITLE
Splunk - enhanced the execution speed of the splunk-search command

### PR DESCRIPTION
**Pull request by guykeller**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/20129

## Description
Splunk's search command used to work relatively slowly.
This problem is now solved.

## Does it break backward compatibility?
   - No

## Must have
- [ ] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

## Technical writer review
Mention and link to the files that require a technical writer review.
- [ ] [YAML file](link)
- [ ] [CHANGELOG](link)

### CODE REVIEW

File: Integrations/SplunkPy/SplunkPy.py [hunk 1]
```diff
@@ -22,6 +23,29 @@
 FETCH_LIMIT = max(min(200, FETCH_LIMIT), 1)
 
 
+class ResponseReaderWrapper(io.RawIOBase):
```
Comment (ronykoz) [inline Integrations/SplunkPy/SplunkPy.py]: add an explanation for this

File: Integrations/SplunkPy/SplunkPy.yml [hunk 2]
```diff
@@ -131,6 +131,16 @@ script:
       name: event_limit
       required: false
       secret: false
+    - name: batch_limit
```
Comment (ronykoz) [inline Integrations/SplunkPy/SplunkPy.yml]: not sure that i get the difference between batch_limit and event limit

File: Integrations/SplunkPy/SplunkPy.py (hunk 1 above)
Comment (guykeller) [reply Integrations/SplunkPy/SplunkPy.py]: Done

File: Integrations/SplunkPy/SplunkPy.py [hunk 3]
```diff
@@ -245,48 +269,73 @@ def request(url, message, **kwargs):
 if demisto.command() == 'splunk-search':
     t = datetime.utcnow() - timedelta(days=7)
     time_str = t.strftime(SPLUNK_TIME_FORMAT)
-    kwargs_oneshot = {"earliest_time": time_str}  # type: Dict[str,Any]
+    kwargs_normalsearch = {
+        "earliest_time": time_str,
+        "exec_mode": "blocking"
```
[the remark this answers is not in the document]
Comment (guykeller) [reply Integrations/SplunkPy/SplunkPy.py]: Done

File: Integrations/SplunkPy/SplunkPy.py [hunk 4]
```diff
@@ -245,48 +269,73 @@ def request(url, message, **kwargs):
...
     if demisto.get(demisto.args(), 'earliest_time'):
-        kwargs_oneshot['earliest_time'] = demisto.args()['earliest_time']
+        kwargs_normalsearch['earliest_time'] = demisto.args()['earliest_time']
     if demisto.get(demisto.args(), 'latest_time'):
-        kwargs_oneshot['latest_time'] = demisto.args()['latest_time']
-    if demisto.get(demisto.args(), 'event_limit'):
-        kwargs_oneshot['count'] = int(demisto.args()['event_limit'])
+        kwargs_normalsearch['latest_time'] = demisto.args()['latest_time']
+
     searchquery_oneshot = demisto.args()['query']
```
[the remark this answers is not in the document]
Comment (guykeller) [reply Integrations/SplunkPy/SplunkPy.py]: It's an alternative for blocking. Fixed naming

File: Integrations/SplunkPy/SplunkPy.yml (hunk 2 above)
Comment (guykeller) [reply Integrations/SplunkPy/SplunkPy.yml]: It's helping performance in some way, let's discuss

**Comment by guykeller** [comment]
@ronykoz It isn't breaking BC as far as I'm aware, and it shouldn't. Could you please explain how it's breaking BC?

**Comment by kirbles19** [comment]
@guykeller Done.

**Comment by ronykoz** [comment]
@guykeller just want to verify that this change is not really changing the logic of getting new incidents, plus i still don't understand the difference between event and batch limit

**Pull request merged**